src/components/steps/general-info/Funders.tsx

```diff
@@ -27,32 +40,78 @@ const Funders: React.FC<FundersProps> = ({ project, onChange }) => {
+    ) => {
+        const newProject = project.set(fieldName, selected);
+        onChange(newProject);
+    };
+
     return (
-        <CardContent style={{ padding: "5px 0 0 0" }}>
-            <Title style={{ marginTop: 35 }}>{getProjectFieldName("funders")}</Title>
-            <div data-test-selector="funders" style={{ paddingRight: 40 }}>
-                <MultiSelector
-                    d2={d2}
-                    searchFilterLabel={true}
-                    ordered={false}
-                    height={300}
-                    onChange={(selected: string[]) =>
-                        onUpdateField("funders", funderOptions, selected)
-                    }
-                    options={funderOptions}
-                    selected={project.funders.map(funder => funder.id)}
+        <>
+            <CardContent style={{ padding: "5px 0 0 0" }}>
+                <Title style={{ marginTop: 35 }}>{getProjectFieldName("funders")}</Title>
+                <div data-test-selector="funders" style={{ paddingRight: 40 }}>
+                    <MultiSelector
+                        d2={d2}
+                        searchFilterLabel={true}
+                        ordered={false}
+                        height={300}
+                        onChange={(selected: string[]) => {
+                            onUpdateField("funders", funderOptions, selected);
+                            setDialogOpen(true);
+                        }}
+                        options={funderOptions}
+                        selected={project.funders.map(funder => funder.id)}
+                    />
+                </div>
+            </CardContent>
+            {isDialogOpen && (
+                <ConfirmationDialog
+                    isOpen={isDialogOpen}
+                    title={i18n.t(
+                        "Would you like to use the funder code in the additional designation field?"
+                    )}
+                    cancelText={i18n.t("No")}
+                    saveText={i18n.t("Yes")}
+                    onCancel={() => {
+                        setAutomaticFill(false);
+                        setDialogOpen(false);
+                    }}
+                    onSave={() => {
+                        setAutomaticFill(true);
+                        if (automaticFill) {
+                            const additionalDesignation = value.reduce(
+                                (accumulator, { shortName }) =>
+                                    accumulator ? `${accumulator}-${shortName}` : shortName,
+                                ""
+                            );
+                            updateAdditionalDesignationField("additional", additionalDesignation);
+                        }
+                        setDialogOpen(false);
+                    }}
+                    maxWidth="sm"
+                    fullWidth={true}
                 />
-            </div>
-        </CardContent>
+            )}
+        </>
     );
 };
```

Comment:
There are some issues:

- Functionally: 
  - Note that when saying "yes" to add the code, it concatenates all the selected funder codes, not the latest added (which I understand is the only one to be added).
  - Problem: Add one funder, say "no". Add another funder, say "yes" -> no code is added. Add another funder, say "yes" -> now all funders are added.

Implementation: The `isDialogOpen/automaticFill/value` infrastructure is too convoluted. Let's do it simpler:
 - We only need one state variable, let's call it `fundersAdded` or something similar, with type `Maybe<Array<{code: string}>>`.  
 - Note that `MultiSelector->onChange` returns all the selected funders, but we are interested only in the newly added. Make a diff between `project.funders` and `selected.` That's the value to set the new state variable.

Reply:
@tokland i haven't been able to figure out how to use one state variable to fix the problem. i have just pushed something that fixes it but i don't know if it's the best solution.